packages/block-library/src/html/index.js

```diff
@@ -56,9 +62,27 @@ export const settings = {
 		],
 	},
 
-	edit: withState( {
-		isPreview: false,
-	} )( ( { attributes, setAttributes, setState, isPreview } ) => (
+	edit: compose( [
+		withSelect( ( select ) => {
+			const { getEditorSettings } = select( 'core/editor' );
+			const styles = getEditorSettings().styles;
+			if ( styles && styles.length > 0 ) {
+				return {
+					styles: map( styles, ( { css, baseURL } ) => {
+						if ( ! baseURL ) {
+							return css;
+						}
+						return traverse( css, urlRewrite( baseURL ) );
+					} ),
+				};
+			}
+
+			return {
+				styles: [],
+			};
+		} ),
```

Comment:
This code is not the fastest one and the main issue is that it will be executed on each subscribe (on each state change).  Should we mirror how it's done in the `EditorProvider` component (when mounting the component)?

Also there's some duplication between these two components. Do you think we can extract the logic somehow?

Reply:
@youknowriad 

Does it mean you need to cache styles?

Reply:
I mean just compute the `styles` prop on initial mount of the component, store it as state or something and avoid computing it on each render.